CRM/Contract/Form/Modify.php

```diff
@@ -10,344 +10,569 @@
+            "unit"     => $this->recurring_contribution["frequency_unit"],
+        ];
+
+        $recurring_contributions = CRM_Contract_RecurringContribution::getAllForContact($contact_id, true);
+
+        $resources = CRM_Core_Resources::singleton();
+
+        $resources->addVars("de.systopia.contract", [
+            "action"                  => $this->modify_action,
+            "cid"                     => $contact_id,
+            "current_amount"          => $this->recurring_contribution["amount"],
+            "current_contract"        => $current_contract,
+            "current_cycle_day"       => $this->recurring_contribution["cycle_day"],
+            "current_frequency"       => $current_frequency,
+            "current_recurring"       => $rc_id,
+            "debitor_name"            => $this->contact["display_name"],
+            "frequencies"             => $frequencies,
+            "grace_end"               => $grace_end,
+            "recurring_contributions" => $recurring_contributions,
+        ]);
+
+        foreach (self::$payment_instruments as $pi_name => $pi_class) {
+            $resources->addVars(
+                "de.systopia.contract/$pi_name",
+                $pi_class::formVars($this->payment_instrument)
+            );
+        }
 
-    // Add the interaction medium
-    foreach(civicrm_api3('Activity', 'getoptions', ['field' => "activity_medium_id", 'options' => ['limit' => 0, 'sort' => 'weight']])['values'] as $key => $value){
-      $mediumOptions[$key] = $value;
     }
-    $this->add('select', 'activity_medium', ts('Source media'), array('' => '- none -') + $mediumOptions, false, array('class' => 'crm-select2'));
 
-    // Add a note field
-    if (version_compare(CRM_Utils_System::version(), '4.7', '<')) {
-      $this->addWysiwyg('activity_details', ts('Notes'), []);
-    } else {
-      $this->add('wysiwyg', 'activity_details', ts('Notes'));
-    }
+    function buildQuickForm () {
 
-    // Then add fields that are dependent on the action
-    if(in_array($this->modify_action, array('update', 'revive'))){
-      $this->addUpdateFields();
-    }elseif($this->modify_action == 'cancel'){
-      $this->addCancelFields();
-    }elseif($this->modify_action == 'pause'){
-      $this->addPauseFields();
-    }
+        // Currency
+        $this->assign("currency", CRM_Sepa_Logic_Settings::defaultCreditor()->currency);
 
-    $this->addButtons([
-      ['type' => 'cancel', 'name' => 'Discard changes', 'submitOnce' => TRUE], // since Cancel looks bad when viewed next to the Cancel action
-      ['type' => 'submit', 'name' => $this->change_class::getChangeTitle(), 'isDefault' => true, 'submitOnce' => TRUE],
-    ]);
+        // Payment change (payment_change)
+        $this->add("select", "payment_change", ts("Payment change"), [
+            "no_change"       => ts("No change"),
+            "create"          => ts("New payment method"),
```

Comment:
When I use this option to create a new SEPA mandate for an existing membership, it actually seems to create two identical mandates. Doesn't seem to apply to "Modify current payment method".